src/Model/Storage.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Kirby\Model;
+
+use Kirby\Data\Data;
+use Kirby\Exception\LogicException;
+use Throwable;
+
+abstract class Storage
+{
+	public function __construct(
+		protected Model $model,
+	) {}
+
+	abstract protected static function contentFile($meta): string;
+
+	public function children(): array
+	{
+		throw new LogicException('This model does not have children');
+	}
+
+	public function files(): array
+	{
+		throw new LogicException('This model does not have files');
+	}
+
+	abstract public static function find(string $class, string $identifier): Model|null;
+
+	public function read(): array
+	{
+		try {
+			return Data::read($this->contentFile($this->model->meta()));
+		} catch (Throwable $e) {
+			return [];
+		}
+	}
+
+	public function changeMeta(array $meta): ModelMeta
+	{
+		$newMeta = clone $this->model->meta();
+
+		foreach ($meta as $key => $value) {
+			$newMeta->$key = $value;
+		}
+
+		foreach (array_keys($meta) as $key) {
+			$this->{'change' . ucfirst($key)}($this->model->meta(), $newMeta);
```

Comment:
`Todo`: We need to detect if each meta prop actually changed. Otherwise we will run a lot of operations that aren't needed. We could do it in each individual action, but that's a lot of duplicated code.

Reply:
Yes, that would indeed be much better.